File: javaexample/src/main/java/io/chthonic/mythos/javaexample/ui/presenters/FusPresenter.java
package io.chthonic.mythos.javaexample.ui.presenters;

import android.os.Bundle;
import android.support.annotation.NonNull;

import io.chthonic.mythos.javaexample.ui.vus.FusVu;

/**
 * Created by jhavatar on 3/3/2016.
 */
public class FusPresenter extends BasePresenter<FusVu> {

    private boolean showRo = true;
    private boolean showDah = true;

    public FusPresenter() {
        super();
    }

    @Override
    public void onLink(@NonNull FusVu vu, Bundle inState, @NonNull Bundle args) {
        super.onLink(vu, inState, args);

        vu.updateRoDisplay(showRo);
        vu.updateDahDisplay(showDah);

        vu.setToggleRoListener(new Runnable() {

            @Override
            public void run() {
                showRo = !showRo;
                FusVu vu = getVu();
                if (vu != null) {
                    vu.updateRoDisplay(showRo);
                }
            }
        });

        vu.setToggleDahListener(new Runnable() {

            @Override
            public void run() {
                showDah = !showDah;
                FusVu vu = getVu();
                if (vu != null) {
                    vu.updateDahDisplay(showDah);
                }
            }
        });

        vu.setText(getText("FUS"));
    }
}

File: kotlinexample/src/main/java/io/chthonic/mythos/kotlinexample/ui/presenters/FusPresenter.kt
package io.chthonic.mythos.kotlinexample.ui.presenters

import android.os.Bundle
import io.chthonic.mythos.kotlinexample.ui.vus.FusVu

/**
 * Created by jhavatar on 3/3/2016.
 */
class FusPresenter : BasePresenter<FusVu>() {
    var showRo: Boolean  = true
    var showDah: Boolean = true

    override fun onLink(vu: FusVu, inState: Bundle?, args: Bundle) {
        super.onLink(vu, inState, args)

        vu.updateRoDisplay(showRo)
        vu.updateDahDisplay(showDah)

        vu.toggleRoListener = {
            showRo = !showRo
            this.vu?.updateRoDisplay(showRo)
        }

        vu.toggleDahListener = {
            showDah = !showDah
            this.vu?.updateDahDisplay(showDah)
        }

        vu.setText(getText("FUS"))
    }
}